pkg/controllers/repositoryserver/suite_test.go

```diff
@@ -15,58 +15,100 @@
-		ErrorIfCRDPathMissing: true,
-		UseExistingCluster:    &useExistingCluster,
-	}
+func (s *RepoServerControllerSuite) SetUpSuite(c *C) {
+	config, err := kube.LoadConfig()
+	c.Assert(err, IsNil)
+	cli, err := kubernetes.NewForConfig(config)
+	c.Assert(err, IsNil)
+	crCli, err := crclientv1alpha1.NewForConfig(config)
+	c.Assert(err, IsNil)
 
-	cfg, err := s.testEnv.Start()
+	// Make sure the CRD's exist.
+	err = resource.CreateCustomResources(context.Background(), config)
+	c.Assert(err, IsNil)
+	err = resource.CreateRepoServerCustomResource(context.Background(), config)
 	c.Assert(err, IsNil)
-	c.Assert(cfg, NotNil)
 
-	err = crkanisteriov1alpha1.AddToScheme(scheme.Scheme)
+	s.kubeCli = cli
+	s.crCli = crCli
+
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:             scheme.Scheme,
+		Port:               9443,
+		MetricsBindAddress: "0",
+	})
 	c.Assert(err, IsNil)
 
-	k8sClient, err := client.New(cfg, client.Options{Scheme: scheme.Scheme})
+	err = (&RepositoryServerReconciler{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr)
 	c.Assert(err, IsNil)
-	c.Assert(k8sClient, NotNil)
-}
 
-func (s *ControllerSuite) TearDownSuite(c *C) {
-	if s.testEnv != nil {
-		c.Log("Tearing down the test environment")
-		err := s.testEnv.Stop()
+	go func() {
+		err = mgr.Start(ctrl.SetupSignalHandler())
 		c.Assert(err, IsNil)
+	}()
+
+	ns := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "repositoryservercontrollertest-",
+		},
 	}
+	cns, err := s.kubeCli.CoreV1().Namespaces().Create(context.Background(), ns, metav1.CreateOptions{})
+	c.Assert(err, IsNil)
+	s.repoServerControllerNamespace = cns.Name
+	s.createRepositoryServerSecrets(c)
+}
+
+func (s *RepoServerControllerSuite) createRepositoryServerSecrets(c *C) {
+	kopiaTLSSecretData, err := getKopiaTLSSecret()
+	c.Assert(err, IsNil)
+	s.repoServerSecrets = repositoryServerSecrets{}
+	s.repoServerSecrets.serverUserAccess, err = createRepositoryServerUserAccessSecret(s.kubeCli, s.repoServerControllerNamespace, getRepoServerUserAccessSecretData("localhost", defaultKopiaRepositoryServerAccessPassword))
+	c.Assert(err, IsNil)
+	s.repoServerSecrets.serverAdmin, err = createRepositoryServerAdminSecret(s.kubeCli, s.repoServerControllerNamespace, getRepoServerAdminSecretData(defaulKopiaRepositoryServerAdminUser, defaultKopiaRepositoryServerAdminPassword))
+	c.Assert(err, IsNil)
+	s.repoServerSecrets.repositoryPassword, err = createRepositoryPassword(s.kubeCli, s.repoServerControllerNamespace, getRepoPasswordSecretData(defaultKopiaRepositoryPassword))
+	c.Assert(err, IsNil)
+	s.repoServerSecrets.serverTLS, err = createKopiaTLSSecret(s.kubeCli, s.repoServerControllerNamespace, kopiaTLSSecretData)
+	c.Assert(err, IsNil)
+	s.repoServerSecrets.storage, err = createStorageLocationSecret(s.kubeCli, s.repoServerControllerNamespace, getDefaultS3CompliantStorageLocation())
```

Comment:
```suggestion
	s.repoServerSecrets.serverUserAccess, err = s.createRepositoryServerUserAccessSecret(getRepoServerUserAccessSecretData("localhost", defaultKopiaRepositoryServerAccessPassword))
	c.Assert(err, IsNil)
	s.repoServerSecrets.serverAdmin, err = s.createRepositoryServerAdminSecret(getRepoServerAdminSecretData(defaulKopiaRepositoryServerAdminUser, defaultKopiaRepositoryServerAdminPassword))
	c.Assert(err, IsNil)
	s.repoServerSecrets.repositoryPassword, err = s.createRepositoryPassword(getRepoPasswordSecretData(defaultKopiaRepositoryPassword))
	c.Assert(err, IsNil)
	s.repoServerSecrets.serverTLS, err = s.createKopiaTLSSecret(kopiaTLSSecretData)
	c.Assert(err, IsNil)
	s.repoServerSecrets.storage, err = s.createStorageLocationSecret(getDefaultS3CompliantStorageLocation())
	c.Assert(err, IsNil)
```
If these utils are not used by other tests, we can keep these under this Suite only and won't have to pass all these arguments.

Reply:
I am moving these functions under pkg/testutils.go so that they can be used for integration test as well